tracer/src/Datadog.Trace.Tools.dd_dotnet/CreatedumpCommand.cs

```diff
@@ -0,0 +1,676 @@
+         || assemblyName.StartsWith("netstandard,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("NLog,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("Npgsql,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("nunit.framework,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("OpenTelemetry,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("OpenTelemetry.Api,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("Oracle.DataAccess,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("Oracle.ManagedDataAccess,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("RabbitMQ.Client,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("Serilog,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("ServiceStack.Redis,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("StackExchange.Redis,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("StackExchange.Redis.StrongName,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("System,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("xunit.execution.desktop,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("xunit.execution.dotnet,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("Yarp.ReverseProxy,", StringComparison.Ordinal))
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    private static bool IsTelemetryEnabled()
+    {
+        var value = Environment.GetEnvironmentVariable(ConfigurationKeys.Telemetry.Enabled);
+
+        if (string.Equals(value, "false", StringComparison.OrdinalIgnoreCase) || value == "0")
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    private static bool IsMethodSuspicious(ClrMethod method)
+    {
+        var assemblyName = Path.GetFileName(method.Type?.Module?.Name ?? string.Empty);
+
+        if (assemblyName.StartsWith("Datadog", StringComparison.OrdinalIgnoreCase))
+        {
+            var typeName = method.Type!.Name;
+
+            if (typeName != null)
+            {
+                if (typeName == "Datadog.Trace.ClrProfiler.AutoInstrumentation.AspNetCore.BlockingMiddleware")
+                {
+                    return false;
+                }
+
+                // Datadog.Trace.ClrProfiler.CallTarget.Handlers.Continuations.TaskContinuationGenerator`4.SyncCallbackHandler.<ContinuationAction>d__4.MoveNext()
+                if (typeName.Contains("TaskContinuationGenerator")
+                 && typeName.Contains("ContinuationAction")
+                 && method.Name == "MoveNext")
+                {
+                    // Very likely to be an async continuation
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        if (method.Type!.Module!.IsDynamic && assemblyName.StartsWith("DuckType"))
```

Comment:
You explicitly acknowledge Type and Module _could_ be `null` at the start, and then assert they can't be here. I'm strongly in favour of removing all the `!` in here and just sucking up the null checks instead 😛

Reply:
Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/5451/commits/ee073bd107bcc3cebfe47566ce399d2a5944e2a9